scripts/wasmtestreport.py

```diff
@@ -622,6 +611,58 @@ def compare_test_results(file1, file2):
     
     return (status, new_failures)
 
+# ----------------------------------------------------------------------
+# Function: run_subprocess
+#
+# Purpose:
+#   Wrapper for subprocess.run with debug logging. 
+#   Improves visibility during unit test execution, and replaces unsafe os.chdir usage by allowing a cwd argument.
+#
+# Variables:
+# - Input:
+#   cmd     : Command to execute (e.g., ["gcc", "main.c"]).
+#   label   : Label used to tag debug log outputs.
+#   cwd     : Working directory for the command.
+#   shell   : Whether to run the command in a shell.
+#   timeout : Timeout for command execution.
+#
+# - Output:
+#   The result of subprocess.run()
+#
+# Raises:
+#   subprocess.TimeoutExpired - If the command exceeds the timeout.
+#   Exception - For all other unexpected execution errors.
+# ----------------------------------------------------------------------
+def run_subprocess(cmd, label="", cwd=None, shell=False, timeout=None):
+    """
+    Wrapper for subprocess.run with optional debug logging.
+    """
+    try:
+        proc = subprocess.run(
+            cmd,
+            capture_output=True,
+            text=True,
+            cwd=cwd,
+            shell=shell,
+            timeout=timeout
+        )
+
+        if DEBUG_MODE:
+            print(f"\n>>> {label.upper()} CMD: {' '.join(map(str, cmd))}")
+            if proc.stdout.strip():
+                print(f"[{label} STDOUT]\n{proc.stdout.strip()}")
+            if proc.stderr.strip():
+                print(f"[{label} STDERR]\n{proc.stderr.strip()}")
+        return proc
+    except subprocess.TimeoutExpired as e:
+        if DEBUG_MODE:
+            print(f"[{label}] TIMEOUT after {timeout}s")
+        raise
+    except Exception as e:
+        if DEBUG_MODE:
+            print(f"[{label}] EXCEPTION: {str(e)}")
+        raise
+
 def main():
     os.chdir(LIND_WASM_BASE)
```

Comment:
I like the general move away from `os.chdir`, and I wonder if this one is still necessary?